### DISCUSSION TITLE
Firebase database as a back-end server for flutter app, the app is working in debug mode for android and IOS. but it is not working in release mode for android only.

### DISCUSSION THREAD

**Discussion by usaikoo** [firebase/flutterfire#6941] [Show and tell]
I am using Firebase database as a back-end server for flutter app, the app is working in debug mode for android and IOS. but it is not working in release mode for android only. IOS is working not only in release or debug mode. Although the code base is same for UI and widget are correct, the android release is not reading the data from Firebase database.

**Comment by CodingDavid8** [comment]
Sounds like you didn't set the internet permission for Android. In Debug mode it is automatically enabled, but for Release mode, you need to add the permission yourself.

Check your AndroidManifest.xml and add this line outside the application tag:
`<uses-permission android:name="android.permission.INTERNET" />`

If you already thought about this, are other internet related services available? Do you get any specific errors?

**Reply by usaikoo** [reply]
Hi thank you for your answer,
I found the solution finally, I found in firebase doc whiich is that I initializate the firebase and validate before going to other pages and it works and also I added

to AndroidManifest.xml not only in main folder but also in debug folder.


```

final Future _initialization = Firebase.initializeApp();
return FutureBuilder(
// Initialize FlutterFire:
future: _initialization,
builder: (context, snapshot) {
// Check for errors
if (snapshot.hasError) {
return Container(child: Text('Error'),);
}

    // Once complete, show your application
    if (snapshot.connectionState == ConnectionState.done) {
      return MaterialApp(
        
        debugShowCheckedModeBanner: false,
        title: 'Exmaple App',
        home: SplashPage(),
      );
    }

    // Otherwise, show something whilst waiting for initialization to complete
    return Loading();
  },
);

```
